### PR TITLE
[SAIC-453] Extend func tests due to different admin tenants migration

### PR DESCRIPTION
**Pull request by pavloivanets**
Extend functional tests to verify migration 
when source and destination migration tenants are different

### CODE REVIEW

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module tests.functional_test
devlab/tests/functional_test.py:15: [W0403(relative-import), ] Relative import 'config', should be 'tests.config'
devlab/tests/functional_test.py:20: [W0403(relative-import), ] Relative import 'generate_load', should be 'tests.generate_load'
devlab/tests/functional_test.py:21: [W0403(relative-import), ] Relative import 'filtering_utils', should be 'tests.filtering_utils'
devlab/tests/functional_test.py:29: [F0401(import-error), ] Unable to import 'cfglib'
devlab/tests/functional_test.py:139: [W0106(expression-not-assigned), FunctionalTest.filter_vms] Expression "[vms.extend(i['vms']) for i in config.tenants if 'vms' in i]" is assigned to nothing
devlab/tests/functional_test.py:163: [W0106(expression-not-assigned), FunctionalTest.filter_images] Expression "[images.append(i['name']) for i in tenant['images']]" is assigned to nothing
devlab/tests/functional_test.py:169: [W0106(expression-not-assigned), FunctionalTest.filter_volumes] Expression "[volumes.extend(i['cinder_volumes']) for i in config.tenants if 'cinder_volumes' in i]" is assigned to nothing
devlab/tests/functional_test.py:193: [W0106(expression-not-assigned), FunctionalTest.get_vms_with_fip_associated] Expression "[vms.extend(i['vms']) for i in config.tenants if 'vms' in i]" is assigned to nothing
************* Module tests.generate_load
devlab/tests/generate_load.py:6: [W0403(relative-import), ] Relative import 'config', should be 'tests.config'
devlab/tests/generate_load.py:8: [W0403(relative-import), ] Relative import 'filtering_utils', should be 'tests.filtering_utils'
devlab/tests/generate_load.py:693: [E1101(no-member), Prerequisites.create_security_group] Instance of 'dict' has no 'id' member
devlab/tests/generate_load.py:758: [E1101(no-member), Prerequisites.create_cinder_volumes] Instance of 'dict' has no 'id' member

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

File: devlab/tests/generate_load.py [hunk 1]
```diff
@@ -75,43 +75,68 @@ class NotFound(Exception):
...
-        self.tenant = os.environ['%s_OS_TENANT_NAME' % cloud_prefix]
-        self.auth_url = os.environ['%s_OS_AUTH_URL' % cloud_prefix]
-        self.image_endpoint = os.environ['%s_OS_IMAGE_ENDPOINT' % cloud_prefix]
-        self.neutron_endpoint = os.environ['%s_OS_NEUTRON_ENDPOINT'
-                                           % cloud_prefix]
+
+        self.cloud_prefix = cloud_prefix.lower()
+        self.configuration_ini = configuration_ini
+
+        if isinstance(configuration_ini, dict):
```
Comment (roman-verchikov) [inline devlab/tests/generate_load.py]: Method arguments must either be of the same type, or follow the same interface, code should not check types of objects it works with.

File: devlab/tests/generate_load.py [hunk 2]
```diff
@@ -75,43 +75,68 @@ class NotFound(Exception):
...
-                                          self.username, self.password,
+        self.cinderclient = cinder.Client(self.username, self.password,
                                           self.tenant, self.auth_url)
         self.openstack_release = self._get_openstack_release()
 
+    def set_tenant(self, tenant):
+        self.tenant = tenant
+
+    def get_tenant(self):
+        return self.tenant
```
Comment (roman-verchikov) [inline devlab/tests/generate_load.py]: Please remove if not used

File: devlab/tests/generate_load.py [hunk 3]
```diff
@@ -75,43 +75,68 @@ class NotFound(Exception):
...
+
+    def get_user(self):
+        return self.config.get(self.cloud_prefix, 'user')
```
Comment (roman-verchikov) [inline devlab/tests/generate_load.py]: Please remove

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module tests.functional_test
devlab/tests/functional_test.py:15: [W0403(relative-import), ] Relative import 'config', should be 'tests.config'
devlab/tests/functional_test.py:20: [W0403(relative-import), ] Relative import 'generate_load', should be 'tests.generate_load'
devlab/tests/functional_test.py:21: [W0403(relative-import), ] Relative import 'filtering_utils', should be 'tests.filtering_utils'
devlab/tests/functional_test.py:29: [F0401(import-error), ] Unable to import 'cfglib'
devlab/tests/functional_test.py:139: [W0106(expression-not-assigned), FunctionalTest.filter_vms] Expression "[vms.extend(i['vms']) for i in config.tenants if 'vms' in i]" is assigned to nothing
devlab/tests/functional_test.py:163: [W0106(expression-not-assigned), FunctionalTest.filter_images] Expression "[images.append(i['name']) for i in tenant['images']]" is assigned to nothing
devlab/tests/functional_test.py:169: [W0106(expression-not-assigned), FunctionalTest.filter_volumes] Expression "[volumes.extend(i['cinder_volumes']) for i in config.tenants if 'cinder_volumes' in i]" is assigned to nothing
devlab/tests/functional_test.py:193: [W0106(expression-not-assigned), FunctionalTest.get_vms_with_fip_associated] Expression "[vms.extend(i['vms']) for i in config.tenants if 'vms' in i]" is assigned to nothing
************* Module tests.generate_load
devlab/tests/generate_load.py:6: [W0403(relative-import), ] Relative import 'config', should be 'tests.config'
devlab/tests/generate_load.py:8: [W0403(relative-import), ] Relative import 'filtering_utils', should be 'tests.filtering_utils'
devlab/tests/generate_load.py:675: [E1101(no-member), Prerequisites.create_security_group] Instance of 'dict' has no 'id' member
devlab/tests/generate_load.py:740: [E1101(no-member), Prerequisites.create_cinder_volumes] Instance of 'dict' has no 'id' member

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by roman-verchikov** [comment]
Looks good, +1

**Comment by roman-verchikov** [comment]
@plomakin please review

**Comment by plomakin** [comment]
Please close this pull request, re-create it whenever gerrit commit and vagrant boxes will be ready.

**Comment by plomakin** [comment]
assuming that gerrit commit will be merged first, this pull request might be merged as well

**Comment by roman-verchikov** [comment]
@pavloivanets please rebase

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module tests.functional_test
devlab/tests/functional_test.py:20: [W0403(relative-import), ] Relative import 'generate_load', should be 'tests.generate_load'
devlab/tests/functional_test.py:165: [W0106(expression-not-assigned), FunctionalTest.filter_volumes] Expression "[volumes.extend(i['cinder_volumes']) for i in config.tenants if 'cinder_volumes' in i]" is assigned to nothing
devlab/tests/functional_test.py:189: [W0106(expression-not-assigned), FunctionalTest.get_vms_with_fip_associated] Expression "[vms.extend(i['vms']) for i in config.tenants if 'vms' in i]" is assigned to nothing
************* Module tests.generate_load
devlab/tests/generate_load.py:12: [E0401(import-error), ] Unable to import 'filtering_utils'
devlab/tests/generate_load.py:24: [W0403(relative-import), ] Relative import 'exception', should be 'tests.exception'
devlab/tests/generate_load.py:691: [E1101(no-member), Prerequisites.create_security_group] Instance of 'dict' has no 'id' member
devlab/tests/generate_load.py:756: [E1101(no-member), Prerequisites.create_cinder_volumes] Instance of 'dict' has no 'id' member

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by ntymtsiv** [comment]
please rebase

**Comment by MirantisWorkloadMobility** [comment]
Can one of the admins verify this patch?

**Comment by MirantisWorkloadMobility** [comment]
Can one of the admins verify this patch?

**Comment by MirantisWorkloadMobility** [comment]
Can one of the admins verify this patch?

**Comment by MirantisWorkloadMobility** [comment]
Can one of the admins verify this patch?

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module tests.functional_test
devlab/tests/functional_test.py:19: [W0403(relative-import), ] Relative import 'generate_load', should be 'tests.generate_load'
devlab/tests/functional_test.py:22: [W0403(relative-import), ] Relative import 'exception', should be 'tests.exception'
devlab/tests/functional_test.py:173: [W0106(expression-not-assigned), FunctionalTest.filter_volumes] Expression "[volumes.extend(i['cinder_volumes']) for i in config.tenants if 'cinder_volumes' in i]" is assigned to nothing
devlab/tests/functional_test.py:197: [W0106(expression-not-assigned), FunctionalTest.get_vms_with_fip_associated] Expression "[vms.extend(i['vms']) for i in config.tenants if 'vms' in i]" is assigned to nothing
************* Module tests.generate_load
devlab/tests/generate_load.py:12: [E0401(import-error), ] Unable to import 'filtering_utils'
devlab/tests/generate_load.py:24: [W0403(relative-import), ] Relative import 'exception', should be 'tests.exception'
devlab/tests/generate_load.py:692: [E1101(no-member), Prerequisites.create_security_group] Instance of 'dict' has no 'id' member
devlab/tests/generate_load.py:757: [E1101(no-member), Prerequisites.create_cinder_volumes] Instance of 'dict' has no 'id' member

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

File: devlab/tests/functional_test.py [hunk 4]
```diff
@@ -45,8 +49,18 @@ class FunctionalTest(unittest.TestCase):
     def __init__(self, *args, **kwargs):
         super(FunctionalTest, self).__init__(*args, **kwargs)
         suppress_dependency_logging()
-        self.src_cloud = Prerequisites(cloud_prefix='SRC', config=config)
-        self.dst_cloud = Prerequisites(cloud_prefix='DST', config=config)
+
+        if not config_ini:
+            raise ConfFileError('Configuration file parameter'
+                                ' --tc-file is missed or '
```
Comment (roman-verchikov) [inline devlab/tests/functional_test.py]: "is missing"

File: devlab/tests/exception.py [hunk 5]
```diff
@@ -16,3 +16,9 @@
 class NotFound(Exception):
     """Raise this exception in case when resource was not found
     """
+
+
+class ConfFileError(RuntimeError):
+    """Raise this exception in case when config file parameter
+       missed or config file format is incorrect
```
Comment (roman-verchikov) [inline devlab/tests/exception.py]: "missing"

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module tests.functional_test
devlab/tests/functional_test.py:19: [W0403(relative-import), ] Relative import 'generate_load', should be 'tests.generate_load'
devlab/tests/functional_test.py:22: [E0401(import-error), ] Unable to import 'exception'
devlab/tests/functional_test.py:173: [W0106(expression-not-assigned), FunctionalTest.filter_volumes] Expression "[volumes.extend(i['cinder_volumes']) for i in config.tenants if 'cinder_volumes' in i]" is assigned to nothing
devlab/tests/functional_test.py:197: [W0106(expression-not-assigned), FunctionalTest.get_vms_with_fip_associated] Expression "[vms.extend(i['vms']) for i in config.tenants if 'vms' in i]" is assigned to nothing
************* Module tests.generate_load
devlab/tests/generate_load.py:12: [E0401(import-error), ] Unable to import 'filtering_utils'
devlab/tests/generate_load.py:24: [W0403(relative-import), ] Relative import 'test_exceptions', should be 'tests.test_exceptions'
devlab/tests/generate_load.py:700: [E1101(no-member), Prerequisites.create_security_group] Instance of 'dict' has no 'id' member
devlab/tests/generate_load.py:765: [E1101(no-member), Prerequisites.create_cinder_volumes] Instance of 'dict' has no 'id' member

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module tests.functional_test
devlab/tests/functional_test.py:19: [W0403(relative-import), ] Relative import 'generate_load', should be 'tests.generate_load'
devlab/tests/functional_test.py:22: [W0403(relative-import), ] Relative import 'test_exceptions', should be 'tests.test_exceptions'
devlab/tests/functional_test.py:171: [W0106(expression-not-assigned), FunctionalTest.filter_volumes] Expression "[volumes.extend(i['cinder_volumes']) for i in config.tenants if 'cinder_volumes' in i]" is assigned to nothing
devlab/tests/functional_test.py:195: [W0106(expression-not-assigned), FunctionalTest.get_vms_with_fip_associated] Expression "[vms.extend(i['vms']) for i in config.tenants if 'vms' in i]" is assigned to nothing
************* Module tests.generate_load
devlab/tests/generate_load.py:12: [E0401(import-error), ] Unable to import 'filtering_utils'
devlab/tests/generate_load.py:24: [W0403(relative-import), ] Relative import 'test_exceptions', should be 'tests.test_exceptions'
devlab/tests/generate_load.py:700: [E1101(no-member), Prerequisites.create_security_group] Instance of 'dict' has no 'id' member
devlab/tests/generate_load.py:765: [E1101(no-member), Prerequisites.create_cinder_volumes] Instance of 'dict' has no 'id' member

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by plomakin** [comment]
+1, let's see how functional tests will finish after that changes.

**Comment by plomakin** [comment]
Please fix this: 
```
+ python generate_load.py
Traceback (most recent call last):
  File "generate_load.py", line 12, in <module>
    from filtering_utils import FilteringUtils
ImportError: No module named filtering_utils
```

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check failed
http://jenkins-cf.vm.mirantis.net:8080/job/cloudferry-unittests-and-style-checks/1734/

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module tests.functional_test
devlab/tests/functional_test.py:19: [W0403(relative-import), ] Relative import 'generate_load', should be 'tests.generate_load'
devlab/tests/functional_test.py:22: [W0403(relative-import), ] Relative import 'test_exceptions', should be 'tests.test_exceptions'
devlab/tests/functional_test.py:171: [W0106(expression-not-assigned), FunctionalTest.filter_volumes] Expression "[volumes.extend(i['cinder_volumes']) for i in config.tenants if 'cinder_volumes' in i]" is assigned to nothing
devlab/tests/functional_test.py:195: [W0106(expression-not-assigned), FunctionalTest.get_vms_with_fip_associated] Expression "[vms.extend(i['vms']) for i in config.tenants if 'vms' in i]" is assigned to nothing
************* Module tests.generate_load
devlab/tests/generate_load.py:23: [W0403(relative-import), ] Relative import 'test_exceptions', should be 'tests.test_exceptions'
devlab/tests/generate_load.py:699: [E1101(no-member), Prerequisites.create_security_group] Instance of 'dict' has no 'id' member
devlab/tests/generate_load.py:764: [E1101(no-member), Prerequisites.create_cinder_volumes] Instance of 'dict' has no 'id' member

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module tests.functional_test
devlab/tests/functional_test.py:19: [W0403(relative-import), ] Relative import 'generate_load', should be 'tests.generate_load'
devlab/tests/functional_test.py:22: [W0403(relative-import), ] Relative import 'test_exceptions', should be 'tests.test_exceptions'
devlab/tests/functional_test.py:171: [W0106(expression-not-assigned), FunctionalTest.filter_volumes] Expression "[volumes.extend(i['cinder_volumes']) for i in config.tenants if 'cinder_volumes' in i]" is assigned to nothing
devlab/tests/functional_test.py:195: [W0106(expression-not-assigned), FunctionalTest.get_vms_with_fip_associated] Expression "[vms.extend(i['vms']) for i in config.tenants if 'vms' in i]" is assigned to nothing
************* Module tests.generate_load
devlab/tests/generate_load.py:23: [W0403(relative-import), ] Relative import 'test_exceptions', should be 'tests.test_exceptions'
devlab/tests/generate_load.py:699: [E1101(no-member), Prerequisites.create_security_group] Instance of 'dict' has no 'id' member
devlab/tests/generate_load.py:764: [E1101(no-member), Prerequisites.create_cinder_volumes] Instance of 'dict' has no 'id' member

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check failed
http://jenkins-cf.vm.mirantis.net:8080/job/cloudferry-unittests-and-style-checks/1737/

**Comment by roman-verchikov** [comment]
retest this please

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module tests.functional_test
devlab/tests/functional_test.py:19: [W0403(relative-import), ] Relative import 'generate_load', should be 'tests.generate_load'
devlab/tests/functional_test.py:22: [W0403(relative-import), ] Relative import 'test_exceptions', should be 'tests.test_exceptions'
devlab/tests/functional_test.py:171: [W0106(expression-not-assigned), FunctionalTest.filter_volumes] Expression "[volumes.extend(i['cinder_volumes']) for i in config.tenants if 'cinder_volumes' in i]" is assigned to nothing
devlab/tests/functional_test.py:195: [W0106(expression-not-assigned), FunctionalTest.get_vms_with_fip_associated] Expression "[vms.extend(i['vms']) for i in config.tenants if 'vms' in i]" is assigned to nothing
************* Module tests.generate_load
devlab/tests/generate_load.py:23: [W0403(relative-import), ] Relative import 'test_exceptions', should be 'tests.test_exceptions'
devlab/tests/generate_load.py:699: [E1101(no-member), Prerequisites.create_security_group] Instance of 'dict' has no 'id' member
devlab/tests/generate_load.py:764: [E1101(no-member), Prerequisites.create_cinder_volumes] Instance of 'dict' has no 'id' member

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module tests.functional_test
devlab/tests/functional_test.py:19: [W0403(relative-import), ] Relative import 'generate_load', should be 'tests.generate_load'
devlab/tests/functional_test.py:22: [W0403(relative-import), ] Relative import 'test_exceptions', should be 'tests.test_exceptions'
devlab/tests/functional_test.py:171: [W0106(expression-not-assigned), FunctionalTest.filter_volumes] Expression "[volumes.extend(i['cinder_volumes']) for i in config.tenants if 'cinder_volumes' in i]" is assigned to nothing
devlab/tests/functional_test.py:195: [W0106(expression-not-assigned), FunctionalTest.get_vms_with_fip_associated] Expression "[vms.extend(i['vms']) for i in config.tenants if 'vms' in i]" is assigned to nothing
************* Module tests.generate_load
devlab/tests/generate_load.py:23: [W0403(relative-import), ] Relative import 'test_exceptions', should be 'tests.test_exceptions'
devlab/tests/generate_load.py:699: [E1101(no-member), Prerequisites.create_security_group] Instance of 'dict' has no 'id' member
devlab/tests/generate_load.py:764: [E1101(no-member), Prerequisites.create_cinder_volumes] Instance of 'dict' has no 'id' member

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by plomakin** [comment]
+1 but first merge gerrit commit

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module tests.functional_test
devlab/tests/functional_test.py:15: [W0403(relative-import), ] Relative import 'config', should be 'tests.config'
devlab/tests/functional_test.py:19: [W0403(relative-import), ] Relative import 'generate_load', should be 'tests.generate_load'
devlab/tests/functional_test.py:20: [W0403(relative-import), ] Relative import 'utils', should be 'tests.utils'
devlab/tests/functional_test.py:22: [W0403(relative-import), ] Relative import 'test_exceptions', should be 'tests.test_exceptions'
devlab/tests/functional_test.py:30: [E0401(import-error), ] Unable to import 'cfglib'
devlab/tests/functional_test.py:171: [W0106(expression-not-assigned), FunctionalTest.filter_volumes] Expression "[volumes.extend(i['cinder_volumes']) for i in config.tenants if 'cinder_volumes' in i]" is assigned to nothing
devlab/tests/functional_test.py:195: [W0106(expression-not-assigned), FunctionalTest.get_vms_with_fip_associated] Expression "[vms.extend(i['vms']) for i in config.tenants if 'vms' in i]" is assigned to nothing
************* Module tests.generate_load
devlab/tests/generate_load.py:8: [W0403(relative-import), ] Relative import 'config', should be 'tests.config'
devlab/tests/generate_load.py:9: [W0403(relative-import), ] Relative import 'utils', should be 'tests.utils'
devlab/tests/generate_load.py:23: [W0403(relative-import), ] Relative import 'test_exceptions', should be 'tests.test_exceptions'
devlab/tests/generate_load.py:699: [E1101(no-member), Prerequisites.create_security_group] Instance of 'dict' has no 'id' member
devlab/tests/generate_load.py:764: [E1101(no-member), Prerequisites.create_cinder_volumes] Instance of 'dict' has no 'id' member

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module tests.functional_test
devlab/tests/functional_test.py:15: [W0403(relative-import), ] Relative import 'config', should be 'tests.config'
devlab/tests/functional_test.py:19: [W0403(relative-import), ] Relative import 'generate_load', should be 'tests.generate_load'
devlab/tests/functional_test.py:20: [W0403(relative-import), ] Relative import 'utils', should be 'tests.utils'
devlab/tests/functional_test.py:23: [W0403(relative-import), ] Relative import 'test_exceptions', should be 'tests.test_exceptions'
devlab/tests/functional_test.py:30: [E0401(import-error), ] Unable to import 'cfglib'
devlab/tests/functional_test.py:171: [W0106(expression-not-assigned), FunctionalTest.filter_volumes] Expression "[volumes.extend(i['cinder_volumes']) for i in config.tenants if ('cinder_volumes' in i) and (not i.get('deleted'))]" is assigned to nothing
devlab/tests/functional_test.py:195: [W0106(expression-not-assigned), FunctionalTest.get_vms_with_fip_associated] Expression "[vms.extend(i['vms']) for i in config.tenants if 'vms' in i]" is assigned to nothing
************* Module tests.generate_load
devlab/tests/generate_load.py:8: [W0403(relative-import), ] Relative import 'config', should be 'tests.config'
devlab/tests/generate_load.py:9: [W0403(relative-import), ] Relative import 'utils', should be 'tests.utils'
devlab/tests/generate_load.py:23: [W0403(relative-import), ] Relative import 'test_exceptions', should be 'tests.test_exceptions'
devlab/tests/generate_load.py:699: [E1101(no-member), Prerequisites.create_security_group] Instance of 'dict' has no 'id' member
devlab/tests/generate_load.py:764: [E1101(no-member), Prerequisites.create_cinder_volumes] Instance of 'dict' has no 'id' member

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module tests.functional_test
devlab/tests/functional_test.py:15: [W0403(relative-import), ] Relative import 'config', should be 'tests.config'
devlab/tests/functional_test.py:19: [W0403(relative-import), ] Relative import 'generate_load', should be 'tests.generate_load'
devlab/tests/functional_test.py:20: [W0403(relative-import), ] Relative import 'utils', should be 'tests.utils'
devlab/tests/functional_test.py:23: [W0403(relative-import), ] Relative import 'test_exceptions', should be 'tests.test_exceptions'
devlab/tests/functional_test.py:31: [E0401(import-error), ] Unable to import 'cfglib'
devlab/tests/functional_test.py:172: [W0106(expression-not-assigned), FunctionalTest.filter_volumes] Expression "[volumes.extend(i['cinder_volumes']) for i in config.tenants if ('cinder_volumes' in i) and (not i.get('deleted'))]" is assigned to nothing
devlab/tests/functional_test.py:196: [W0106(expression-not-assigned), FunctionalTest.get_vms_with_fip_associated] Expression "[vms.extend(i['vms']) for i in config.tenants if 'vms' in i]" is assigned to nothing
************* Module tests.generate_load
devlab/tests/generate_load.py:8: [W0403(relative-import), ] Relative import 'config', should be 'tests.config'
devlab/tests/generate_load.py:9: [W0403(relative-import), ] Relative import 'utils', should be 'tests.utils'
devlab/tests/generate_load.py:23: [W0403(relative-import), ] Relative import 'test_exceptions', should be 'tests.test_exceptions'
devlab/tests/generate_load.py:699: [E1101(no-member), Prerequisites.create_security_group] Instance of 'dict' has no 'id' member
devlab/tests/generate_load.py:764: [E1101(no-member), Prerequisites.create_cinder_volumes] Instance of 'dict' has no 'id' member

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module tests.functional_test
devlab/tests/functional_test.py:15: [W0403(relative-import), ] Relative import 'config', should be 'tests.config'
devlab/tests/functional_test.py:19: [W0403(relative-import), ] Relative import 'generate_load', should be 'tests.generate_load'
devlab/tests/functional_test.py:20: [W0403(relative-import), ] Relative import 'utils', should be 'tests.utils'
devlab/tests/functional_test.py:23: [W0403(relative-import), ] Relative import 'test_exceptions', should be 'tests.test_exceptions'
devlab/tests/functional_test.py:31: [E0401(import-error), ] Unable to import 'cfglib'
devlab/tests/functional_test.py:172: [W0106(expression-not-assigned), FunctionalTest.filter_volumes] Expression "[volumes.extend(i['cinder_volumes']) for i in config.tenants if ('cinder_volumes' in i) and (not i.get('deleted'))]" is assigned to nothing
devlab/tests/functional_test.py:196: [W0106(expression-not-assigned), FunctionalTest.get_vms_with_fip_associated] Expression "[vms.extend(i['vms']) for i in config.tenants if 'vms' in i]" is assigned to nothing
************* Module tests.generate_load
devlab/tests/generate_load.py:8: [W0403(relative-import), ] Relative import 'config', should be 'tests.config'
devlab/tests/generate_load.py:9: [W0403(relative-import), ] Relative import 'utils', should be 'tests.utils'
devlab/tests/generate_load.py:23: [W0403(relative-import), ] Relative import 'test_exceptions', should be 'tests.test_exceptions'
devlab/tests/generate_load.py:699: [E1101(no-member), Prerequisites.create_security_group] Instance of 'dict' has no 'id' member
devlab/tests/generate_load.py:764: [E1101(no-member), Prerequisites.create_cinder_volumes] Instance of 'dict' has no 'id' member
devlab/tests/generate_load.py:1: [W0611(unused-import), ] Unused import os

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module tests.functional_test
devlab/tests/functional_test.py:15: [W0403(relative-import), ] Relative import 'config', should be 'tests.config'
devlab/tests/functional_test.py:19: [W0403(relative-import), ] Relative import 'generate_load', should be 'tests.generate_load'
devlab/tests/functional_test.py:20: [W0403(relative-import), ] Relative import 'utils', should be 'tests.utils'
devlab/tests/functional_test.py:23: [W0403(relative-import), ] Relative import 'test_exceptions', should be 'tests.test_exceptions'
devlab/tests/functional_test.py:31: [E0401(import-error), ] Unable to import 'cfglib'
devlab/tests/functional_test.py:172: [W0106(expression-not-assigned), FunctionalTest.filter_volumes] Expression "[volumes.extend(i['cinder_volumes']) for i in config.tenants if ('cinder_volumes' in i) and (not i.get('deleted'))]" is assigned to nothing
devlab/tests/functional_test.py:196: [W0106(expression-not-assigned), FunctionalTest.get_vms_with_fip_associated] Expression "[vms.extend(i['vms']) for i in config.tenants if 'vms' in i]" is assigned to nothing
************* Module tests.generate_load
devlab/tests/generate_load.py:6: [W0403(relative-import), ] Relative import 'config', should be 'tests.config'
devlab/tests/generate_load.py:7: [W0403(relative-import), ] Relative import 'utils', should be 'tests.utils'
devlab/tests/generate_load.py:21: [W0403(relative-import), ] Relative import 'test_exceptions', should be 'tests.test_exceptions'
devlab/tests/generate_load.py:706: [E1101(no-member), Prerequisites.create_security_group] Instance of 'dict' has no 'id' member
devlab/tests/generate_load.py:771: [E1101(no-member), Prerequisites.create_cinder_volumes] Instance of 'dict' has no 'id' member

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success
http://jenkins-cf.vm.mirantis.net:8080/job/cloudferry-unittests-and-style-checks/1806/

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by plomakin** [comment]
+1, assuming that gerrit commit will be merged first, this pull request might be merged as well

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module tests.functional_test
devlab/tests/functional_test.py:15: [W0403(relative-import), ] Relative import 'config', should be 'tests.config'
devlab/tests/functional_test.py:19: [W0403(relative-import), ] Relative import 'generate_load', should be 'tests.generate_load'
devlab/tests/functional_test.py:20: [W0403(relative-import), ] Relative import 'utils', should be 'tests.utils'
devlab/tests/functional_test.py:23: [W0403(relative-import), ] Relative import 'test_exceptions', should be 'tests.test_exceptions'
devlab/tests/functional_test.py:31: [E0401(import-error), ] Unable to import 'cfglib'
devlab/tests/functional_test.py:172: [W0106(expression-not-assigned), FunctionalTest.filter_volumes] Expression "[volumes.extend(i['cinder_volumes']) for i in config.tenants if ('cinder_volumes' in i) and (not i.get('deleted'))]" is assigned to nothing
devlab/tests/functional_test.py:196: [W0106(expression-not-assigned), FunctionalTest.get_vms_with_fip_associated] Expression "[vms.extend(i['vms']) for i in config.tenants if 'vms' in i]" is assigned to nothing
************* Module tests.generate_load
devlab/tests/generate_load.py:6: [W0403(relative-import), ] Relative import 'config', should be 'tests.config'
devlab/tests/generate_load.py:7: [W0403(relative-import), ] Relative import 'utils', should be 'tests.utils'
devlab/tests/generate_load.py:21: [W0403(relative-import), ] Relative import 'test_exceptions', should be 'tests.test_exceptions'
devlab/tests/generate_load.py:706: [E1101(no-member), Prerequisites.create_security_group] Instance of 'dict' has no 'id' member
devlab/tests/generate_load.py:771: [E1101(no-member), Prerequisites.create_cinder_volumes] Instance of 'dict' has no 'id' member

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by plomakin** [comment]
+1, assuming that gerrit commit will be merged first

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module tests.functional_test
devlab/tests/functional_test.py:15: [W0403(relative-import), ] Relative import 'config', should be 'tests.config'
devlab/tests/functional_test.py:19: [W0403(relative-import), ] Relative import 'generate_load', should be 'tests.generate_load'
devlab/tests/functional_test.py:20: [W0403(relative-import), ] Relative import 'utils', should be 'tests.utils'
devlab/tests/functional_test.py:23: [W0403(relative-import), ] Relative import 'test_exceptions', should be 'tests.test_exceptions'
devlab/tests/functional_test.py:31: [E0401(import-error), ] Unable to import 'cfglib'
devlab/tests/functional_test.py:172: [W0106(expression-not-assigned), FunctionalTest.filter_volumes] Expression "[volumes.extend(i['cinder_volumes']) for i in config.tenants if ('cinder_volumes' in i) and (not i.get('deleted'))]" is assigned to nothing
devlab/tests/functional_test.py:196: [W0106(expression-not-assigned), FunctionalTest.get_vms_with_fip_associated] Expression "[vms.extend(i['vms']) for i in config.tenants if 'vms' in i]" is assigned to nothing
************* Module tests.generate_load
devlab/tests/generate_load.py:6: [W0403(relative-import), ] Relative import 'config', should be 'tests.config'
devlab/tests/generate_load.py:7: [W0403(relative-import), ] Relative import 'utils', should be 'tests.utils'
devlab/tests/generate_load.py:21: [W0403(relative-import), ] Relative import 'test_exceptions', should be 'tests.test_exceptions'
devlab/tests/generate_load.py:706: [E1101(no-member), Prerequisites.create_security_group] Instance of 'dict' has no 'id' member
devlab/tests/generate_load.py:771: [E1101(no-member), Prerequisites.create_cinder_volumes] Instance of 'dict' has no 'id' member

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by roman-verchikov** [comment]
can be merged

**Pull request merged**